first_order_model/shrink_util.py

```diff
@@ -1061,52 +1126,54 @@ def shrink_model(model_copy, layer_graph, layer, count, sort):
         model_copy = channel_prune(model_copy, deletions)
     return model_copy
 
+
 def try_reduce(curr_loss, curr_model, dataloader, layer_graph,
                layer, kp_detector, discriminator, train_params, model, target,
                current, lr_size, generator_type, metrics_dataloader,
-               generator_full, sort, steps_per_it, device_ids, log_dir, discriminator_full):
+               generator_full, sort, steps_per_it, device_ids, log_dir, discriminator_full, image_shape):
     """
-    Most complicated function.
     High level summary:
-    Try to shrink the model to the target size.
+    Try to shrink the model to the target size by deleting layer (passed in as an argument)
     Train the model for a short term fine tune
     Test the model on the metrics dataset
-    If the model beats the current best model, return it. If at any point it fails, return None, None
+    If the model beats the current best model, return it. If at any point it fails, return Nones
+    This function is called ~50 times, with each input layer as an argument once in one iteration of netadapt.
     """
     with torch.no_grad():
         model_copy = copy.deepcopy(model)
 
     # Ignore this layer if it has only one output, or its output is never used
-    if 1 >= layer_graph[layer].o:
+    if 1 >= layer_graph[layer].out_channels:
         return None, None, None, None
     if len(layer_graph[layer].after) == 0:
         return None, None, None, None
 
     # Reduce the layer by size 1 to check how much it affects model size.
     model_copy = shrink_model(model_copy, layer_graph, layer, 1, sort)
 
-
-    after_1_reduce = get_model_macs(log_dir, model_copy, kp_detector, torch.device('cuda' if torch.cuda.is_available() else 'cpu'), lr_size, 512, 2)
+    after_1_reduce = get_decode_and_bottleneck_macs(log_dir, model_copy, kp_detector, torch.device('cuda' if torch.cuda.is_available() else 'cpu'), lr_size, image_shape, 2)
```

Comment:
"macs_after_removing_one_channel" - currently unclear what you're tracking after 1 reduce

Reply:
1 reduce is also not descriptive

Reply:
Fixed